### DISCUSSION TITLE
Electromagnetics Kernel for Motor Design

### DISCUSSION THREAD

**Discussion by mrplants** [idaholab/moose#23601] [Q&A Modules: Electromagnetics] [answered]
Is it possible to use Moose builtin Physics kernels to analyze electric motor/generator designs?  I skimmed the [Electromagnetics Module](https://mooseframework.inl.gov/modules/electromagnetics/index.html) and it seemed focused solely on wave propagation for antenna design, RF, etc.  Is there a kernel that analyzes electromagnetic forces from interacting magnetic fields?

If not, any tips on how to put one together?  I'm not a PhD, but can handle my PDEs if absolutely necessary.

**Comment by GiudGiud** [comment]
@cticenhour 
Can you please comment on this?
We re not far off from modeling coupled EM mechanics right?

**Comment by cticenhour** [comment, marked as answer]
@GiudGiud was correct, we are interested in developing coupled electro-mechanical capabilities between the electromagnetics module and the tensor (solid) mechanics module, which would involve the creation of an electromagnetic force kernel as a source term in a larger mechanics problem. We may have some funding later in the fall to start performing this work as part of another project, but that is not guaranteed yet. If you wanted to tackle it yourself, feel free to reach out to me; I created the electromagnetic module and am its current lead. 

As far as tips, since I don't know your background - "Introduction to Electrodynamics" by David Griffiths would be a useful book to get started learning the base equations and can be easily found online. Griffith's book talks some about the Lorentz force, which describes the electromagnetic force on a given particle, but in this case we could want the Laplace force. This is the same idea, but described and manifested at the level of currents instead of individual charges. Stepping out from Griffith's book, I have a somewhat harder time passing along a comprehensive single text, but searching around there is an older text on MIT OpenCourseWare on Electromechanical Dynamics that might be useful: [Part 1 - Discrete Systems](https://ocw.mit.edu/ans7870/resources/woodson/textbook/emd_part1.pdf), [Part 2 - Fields, Forces, and Motion](https://ocw.mit.edu/ans7870/resources/woodson/textbook/emd_part2.pdf), and [Part 3 - Elastic and Fluid Media](https://ocw.mit.edu/ans7870/resources/woodson/textbook/emd_part3.pdf). This text is an unknown to me, but did look very wide-ranging at a skim!

Regarding tips for working with MOOSE and the modules, we have a large number of [examples and tutorials](https://mooseframework.inl.gov/getting_started/examples_and_tutorials/index.html) that you might want to check out. The main workshop example as well as some of the guided individual examples would show how to make new kernels and objects. 

Hopefully this helps get started looking at the problem! Again, feel free to reach out again if you wish to discuss this more.

**Reply by mrplants** [reply]
Thank you!
